voc/python/ast.py

```diff
@@ -1809,13 +1827,37 @@ def visit_Yield(self, node):
         )
 
         for var, index in self.context.local_vars.items():
-            if index is not None and var not in ('<generator>', '#locals'):
+            if index is not None and var not in ('<generator>', '#locals') \
+                    and "#exception-" not in var:  # Don't load exception
                 self.context.add_opcodes(
                     ALOAD_name('#locals'),
                     java.Map.get(var),
                     JavaOpcodes.ASTORE(index),
                 )
 
+        if hasattr(node, "regular_yield"):
+            # reset message to None
+            self.context.load_name('<generator>')
+            self.context.add_opcodes(
+                JavaOpcodes.INVOKEVIRTUAL(
+                    'org/python/types/Generator',
+                    'reset_message',
+                    args=[],
+                    returns='V'
+                )
+            )
+
+        # throw exception if there is one
```

Comment:
The comment says "*if* there is one" - but this code is executed regardless. It would be worth noting (both here, and in a docstring in the Java code) that `throw_exception` is a no-op if the generator doesn't have an exception state.

Reply:
Resolved.